source/agent/analytics/index.js

```diff
@@ -95,65 +100,186 @@ module.exports = function (rpcClient, rpcId, agentId, clusterIp) {
+
+        this.connectionclose = () => {
+            destroyStream(options.controller, newStreamId);
+        }
+        inputs[connectionId] = true;
+
+        engine.addEventListener('fatal', function (error) {
+            log.error('GStreamer pipeline error:', error);
+            notifyStatus(options.controller, connectionId, 'out', {type: 'failed', reason: 'Analytics error: ' + error});
+        });
+
+        const streamInfo = {
+          type: 'analytics',
+          media: {video: Object.assign({}, videoFormat, videoParameters)},
+          analyticsId: connectionId,
+          locality: {agent:agentId, node:rpcId},
+        };
+        engine.addEventListener('streamadded', function (data) {
+            log.debug('GStreamer pipeline stream generated:', data);
+
+            var dispatcher = new MediaFrameMulticaster();
+            if (engine.addOutput(dispatcher)){
+              outputs[connectionId] = {dispatcher: dispatcher};
+              router.addLocalSource(newStreamId, connectionType, dispatcher.source());
+              callback('callback', 'ok');
+            } else {
+              callback('callback', 'error', 'Failed in adding output')
+            }
+
+            var params;
+              try {
+              params = JSON.parse(data)
+              streamInfo.media.video.bitrate = bitrate;
+              if (params.codec) {
+                streamInfo.media.video.codec = params.codec;
+              }
+
+              if (params.profile) {
+                if (params.profile === "constrained-baseline") {
+                  streamInfo.media.video.profile = 'CB';
+                } else if (params.profile === "baseline") {
+                  streamInfo.media.video.profile = 'B';
+                } else if (params.profile === "main") {
+                  streamInfo.media.video.profile = 'M';
+                } else if (params.profile === "high") {
+                  streamInfo.media.video.profile = 'H';
+                } else {
+                  log.error("Not supported profile:", params.profile);
+                  return;
+                }
+              }
+              generateStream(options.controller, newStreamId, streamInfo);
+            } catch (e) {
+              log.error("Parse stream added data with error:", e);
+            }
+        });
+
+        return callback('callback', 'ok');
+      } else {
+        callback('callback', 'error', 'Connection already exist: '  + connectionId);
+      }
+    };
+
+    that.unsubscribe = function (connectionId, callback) {
+        log.debug('unsubscribe, connectionId:', connectionId);
+    };
+
+    // streamInfo = {id: 'string', ip: 'string', port: 'number'}
+    // from = {audio: streamInfo, video: streamInfo, data: streamInfo}
+    that.linkup = function (connectionId, fromInfo, callback) {
+        log.debug('linkup, connectionId:', connectionId, 'from:', fromInfo);
+        if (inputs[connectionId]) {
```

Comment:
inputs[connectionId] share  the same engine?
Only 1 engine for multiple inputs? or just 1 input for 1 process?

Reply:
only 1 input for 1 process